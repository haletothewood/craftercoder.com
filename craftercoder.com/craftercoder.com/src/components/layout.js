import React from "react"
import PropTypes from "prop-types"

import Header from "./header"
import NavBar from "./navbar"

import { useSiteMetadata } from "../hooks/useSiteMetaData"

import "../styles/layout.scss"

export const PureLayout = ({ site, children }) => {
  return (
    <div id="app">
      <NavBar pageWrapId={"page-wrap"} outerContainerId={"app"} />
      <Header siteTitle={site.title} />
      <div
        style={{
          margin: `0 auto`,
          maxWidth: 960,
          padding: `0px 1.0875rem 1.45rem`,
          paddingTop: 0,
        }}
      >
        <main
          style={{
            paddingTop: 120,
          }}
          id="page-wrap"
        >
          {children}
        </main>
        <footer>
          Â© {new Date().getFullYear()}, Built with
          {` `}
          <a href="https://www.gatsbyjs.org">Gatsby</a>
          {` `}
          by <a href="https://davidhalewood.com">David Halewood</a>
        </footer>
      </div>
    </div>
  )
}

PureLayout.propTypes = {
  site: PropTypes.object,
  children: PropTypes.array,
}

export const Layout = props => {
  const site = useSiteMetadata()
  return <PureLayout {...props} site={site} />
}

export default Layout
