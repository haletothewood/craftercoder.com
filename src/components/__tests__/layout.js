import React from "react"
import renderer from "react-test-renderer"
import { PureLayout as Layout } from "../layout"

const site = {
  title: `Crafter Coder`,
}

describe("Layout", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Layout site={site} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
