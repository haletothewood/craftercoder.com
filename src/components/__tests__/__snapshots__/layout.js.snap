// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  id="app"
>
  <div>
    <div
      className="bm-overlay"
      onClick={[Function]}
      style={
        Object {
          "MozTransform": "translate3d(100%, 0, 0)",
          "MsTransform": "translate3d(100%, 0, 0)",
          "OTransform": "translate3d(100%, 0, 0)",
          "WebkitTransform": "translate3d(100%, 0, 0)",
          "background": "rgba(0, 0, 0, 0.3)",
          "height": "100%",
          "opacity": 0,
          "position": "fixed",
          "transform": "translate3d(100%, 0, 0)",
          "transition": "opacity 0.3s, transform 0s 0.3s",
          "width": "100%",
          "zIndex": 1000,
        }
      }
    />
    <div
      className="bm-menu-wrap"
      id=""
      style={
        Object {
          "MozTransform": "translate3d(100%, 0, 0)",
          "MsTransform": "translate3d(100%, 0, 0)",
          "OTransform": "translate3d(100%, 0, 0)",
          "WebkitTransform": "translate3d(100%, 0, 0)",
          "height": "100%",
          "position": "fixed",
          "right": 0,
          "transform": "translate3d(100%, 0, 0)",
          "transition": "all 0.5s",
          "width": "100%",
          "zIndex": 1100,
        }
      }
    >
      <div
        className="bm-menu"
        style={
          Object {
            "boxSizing": "border-box",
            "height": "100%",
            "overflow": "auto",
          }
        }
      >
        <nav
          className="bm-item-list"
          style={
            Object {
              "height": "100%",
            }
          }
        >
          <a
            className="bm-item menu-item"
            href="/"
            id="home"
            style={
              Object {
                "display": "block",
              }
            }
            tabIndex={-1}
          >
            Home
          </a>
          <a
            className="bm-item menu-item"
            href="/about"
            id="about"
            style={
              Object {
                "display": "block",
              }
            }
            tabIndex={-1}
          >
            About
          </a>
          <a
            className="bm-item menu-item"
            href="/posts"
            id="posts"
            style={
              Object {
                "display": "block",
              }
            }
            tabIndex={-1}
          >
            Posts
          </a>
          <a
            className="bm-item menu-item"
            href="/resources"
            id="resources"
            style={
              Object {
                "display": "block",
              }
            }
            tabIndex={-1}
          >
            Resources
          </a>
          <a
            className="bm-item menu-item"
            href="/tags"
            id="tags"
            style={
              Object {
                "display": "block",
              }
            }
            tabIndex={-1}
          >
            Tags
          </a>
        </nav>
      </div>
      <div
        style={Object {}}
      >
        <div
          className="bm-cross-button"
          style={
            Object {
              "height": 24,
              "position": "absolute",
              "right": 8,
              "top": 8,
              "width": 24,
            }
          }
        >
          <span
            style={
              Object {
                "position": "absolute",
                "right": "14px",
                "top": "6px",
              }
            }
          >
            <span
              className="bm-cross"
              style={
                Object {
                  "height": 14,
                  "position": "absolute",
                  "transform": "rotate(45deg)",
                  "width": 3,
                }
              }
            />
            <span
              className="bm-cross"
              style={
                Object {
                  "height": 14,
                  "position": "absolute",
                  "transform": "rotate(-45deg)",
                  "width": 3,
                }
              }
            />
          </span>
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "transparent",
                "border": "none",
                "cursor": "pointer",
                "fontSize": 0,
                "height": "100%",
                "left": 0,
                "margin": 0,
                "padding": 0,
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            Close Menu
          </button>
        </div>
      </div>
    </div>
    <div
      style={Object {}}
    >
      <div
        className="bm-burger-button"
        style={
          Object {
            "zIndex": 1000,
          }
        }
      >
        <span>
          <span
            className="bm-burger-bars"
            style={
              Object {
                "height": "20%",
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": "0%",
              }
            }
          />
          <span
            className="bm-burger-bars"
            style={
              Object {
                "height": "20%",
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": "40%",
              }
            }
          />
          <span
            className="bm-burger-bars"
            style={
              Object {
                "height": "20%",
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": "80%",
              }
            }
          />
        </span>
        <button
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "background": "transparent",
              "border": "none",
              "cursor": "pointer",
              "fontSize": 0,
              "height": "100%",
              "left": 0,
              "margin": 0,
              "padding": 0,
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
        >
          Open Menu
        </button>
      </div>
    </div>
  </div>
  <header
    style={
      Object {
        "background": "#282828",
        "borderBottom": "solid 2px #1d1d1d",
        "marginBottom": "1.45rem",
        "position": "fixed",
        "width": "100%",
        "zIndex": 1,
      }
    }
  >
    <div
      style={
        Object {
          "margin": "0 auto",
          "maxWidth": 960,
          "padding": "1.45rem 1.0875rem",
        }
      }
    >
      <h1
        style={
          Object {
            "margin": 0,
          }
        }
      >
        <a
          href="/"
          style={
            Object {
              "color": "#d79921",
              "textDecoration": "none",
            }
          }
        >
          Crafter Coder
        </a>
      </h1>
    </div>
  </header>
  <div
    style={
      Object {
        "margin": "0 auto",
        "maxWidth": 960,
        "padding": "0px 1.0875rem 1.45rem",
        "paddingTop": 0,
      }
    }
  >
    <main
      id="page-wrap"
      style={
        Object {
          "paddingTop": 120,
        }
      }
    />
    <footer>
      Â© 
      2020
      , Built with
       
      <a
        href="https://www.gatsbyjs.org"
      >
        Gatsby
      </a>
       
      by 
      <a
        href="https://davidhalewood.com"
      >
        David Halewood
      </a>
    </footer>
  </div>
</div>
`;
